independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/rest/RestClient.java

```diff
@@ -0,0 +1,154 @@
+package io.quarkus.analytics.rest;
+
+import static io.quarkus.analytics.util.StringUtils.getObjectMapper;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import io.quarkus.analytics.dto.config.AnalyticsRemoteConfig;
+import io.quarkus.analytics.dto.config.Identity;
+import io.quarkus.analytics.dto.config.RemoteConfig;
+import io.quarkus.analytics.dto.segment.Track;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+
+/**
+ * Client to post the analytics data to the upstream collection tool.
+ * We use plain REST API calls and not any wrapping library.
+ */
+public class RestClient implements ConfigClient, SegmentClient {
+
+    static final String IDENTITY_ENDPOINT = "v1/identify";
+    static final String TRACK_ENDPOINT = "v1/track";
+    static final int TIMEOUT = 1000;// milliseconds
+    public static final int TTL = TIMEOUT * 2;// milliseconds
+    static final URI CONFIG_URI = getUri(
+            "https://raw.githubusercontent.com/brunobat/tests/main/03656937-19FD-4C83-9066-C76631D445EA");//FIXME config location
+    private static final String AUTH_HEADER = getAuthHeader("SGGi49IwHoDEpE4NVBEHJDZ4uyzeoI4M"); //FIXME dev key
+    private static final int SEGMENT_POST_RESPONSE_CODE = 200; // sad but true
+
+    static URI getUri(final String uri) {
+        try {
+            return new URI(uri);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * All with the same authentication
+     *
+     * @return BAsic Auth Header value
+     * @param key
+     */
+    static String getAuthHeader(final String key) {
+        final String auth = key + ":";
+        return "Basic " + Base64.getEncoder().encodeToString(
+                auth.getBytes(StandardCharsets.ISO_8859_1));
+    }
+
+    private final MessageWriter log;
+
+    private final URI segmentIdentityUri;
+
+    private final URI segmentTraceUri;
+
+    public RestClient(MessageWriter log) {
+        this.log = log;
+        final String segmentBaseUri = System.getProperty("quarkus.analytics.uri.base", "https://api.segment.io/");
```

Comment:
At least in Gradle, only the first Gradle build invocation will query the system property via the `AnalyticsService` singleton. Can we instantiate `AnalyticsService` for every Gradle/Maven project build so this becomes dynamic and can be changed for every build tool invocation?

Reply:
sure

Reply:
AnalyticsService is now instantiated on every run.